lib/charms/bind/v0/dns_record.py

```diff
@@ -0,0 +1,570 @@
+    INVALID_DATA = "invalid_data"
+    FAILURE = "failure"
+    UNKNOWN = "unknown"
+    PENDING = "pending"
+
+    @classmethod
+    def _missing_(cls, _: object) -> "Status":
+        """Handle the enum when the value is missing.
+
+        Returns:
+            value: Status.UNKNOWN.
+        """
+        return cls(cls.UNKNOWN)
+
+
+class RecordType(str, Enum):
+    """Represent the DNS record types.
+
+    Attributes:
+        A: A
+        AAAA: AAAA
+        CNAME: CNAME
+        MX: MX
+        DKIM: DKIM
+        SPF: SPF
+        DMARC: DMARC
+        TXT: TXT
+        CAA: CAA
+        SRV: SRV
+        SVCB: SVCB
+        HTTPS: HTTPS
+        PTR: PTR
+        SOA: SOA
+        NS: NS
+        DS: DS
+        DNSKEY: DNSKEY
+    """
+
+    A = "A"
+    AAAA = "AAAA"
+    CNAME = "CNAME"
+    MX = "MX"
+    DKIM = "DKIM"
+    SPF = "SPF"
+    DMARC = "DMARC"
+    TXT = "TXT"
+    CAA = "CAA"
+    SRV = "SRV"
+    SVCB = "SVCB"
+    HTTPS = "HTTPS"
+    PTR = "PTR"
+    SOA = "SOA"
+    NS = "NS"
+    DS = "DS"
+    DNSKEY = "DNSKEY"
+
+
+class RecordClass(str, Enum):
+    """Represent the DNS record classes.
+
+    Attributes:
+        IN: IN
+    """
+
+    IN = "IN"
+
+
+class DNSProviderData(BaseModel):
+    """Represent the DNS provider data.
+
+    Attributes:
+        uuid: UUID for the domain request.
+        status: status for the domain request.
+        description: status description for the domain request.
+    """
+
+    uuid: UUID
+    status: Status
+    description: Optional[str] = None
+
+
+class DNSRecordProviderData(BaseModel):
+    """List of domains for the provider to manage.
+
+    Attributes:
+        dns_domains: list of domains to manage.
+        dns_entries: list of entries to manage.
+    """
+
+    dns_domains: List[DNSProviderData] = Field(min_length=1)
+    dns_entries: List[DNSProviderData] = Field(min_length=1)
+
+    def to_relation_data(self) -> Dict[str, str]:
+        """Convert an instance of DNSRecordProviderData to the relation representation.
+
+        Returns:
+            Dict containing the representation.
+        """
+        dumped_model = self.model_dump(exclude_unset=True)
+        dumped_data = {}
+        for key, value in dumped_model.items():
+            dumped_data[key] = json.dumps(value, default=str)
+        return dumped_data
+
+    @classmethod
+    def from_relation_data(cls, relation_data: Dict[str, str]) -> "DNSRecordProviderData":
+        """Initialize a new instance of the DNSRecordProviderData class from the relation data.
+
+        Args:
+            relation_data: the relation data.
+
+        Returns:
+            A DNSRecordProviderData instance.
+
+        Raises:
+            ValidationError if the value is not parseable.
+        """
+        print(relation_data)
+        try:
+            loaded_data = {}
+            for key, value in relation_data.items():
+                loaded_data[key] = json.loads(value)
+            return cls(**loaded_data)
```

Comment:
Well instead of using the unpack operator here I'd do:
`return DNSRecordProviderData.model_validate(loaded_data)`